### DISCUSSION TITLE
Have I misunderstood? Model Find returning data that doesn't belong to that model.

### DISCUSSION THREAD

**Discussion by AlexStansfield** [sensedeep/dynamodb-onetable#475] [General]
I'm not sure if this is the same issues as described here https://github.com/sensedeep/dynamodb-onetable/discussions/421.

However I'm trying to understand how OneTable handles a specific use case as it's not working as I had expected.

I have a table that has an Order model and an OrderItem model. 

The Primary hash is `pk` and the Primary sort is `sk`. With these parts of the schema:

Order:
```
      pk: {
        type: String,
        value: 'Company#${companyId}',
      },
      sk: {
        type: String,
        value: 'Order#${id}',
      },
```

Order Item
```
      pk: {
        type: String,
        value: 'Company#${companyId}',
      },
      sk: {
        type: String,
        value:
          'Order#${orderId}:OrderItem#${id}',
      },
```

The intention was this:
1. I can find Orders belonging to a company by doing a `model.find` with just the `companyId`
2. I can find an order and all it's items by doing `table.queryItems` by giving the `pk` and also the `sk` with a "begins" and using `groupByType` to get the Order model and the OrderItem models (as described here - https://doc.onetable.io/api/table/examples/)

The 2nd one seems to work fine. 

But when I try the first it's giving me all the Order Items too, the Order items have their fields that are not in an Order stripped out, but are clearly Order items as they are missing fields that an Order has and an Order item doesn't

I had expected as I called the `find` on the model that it would check the `_type` field to make sure what it was returning was of the correct model and strip out anything else.

Have I misunderstood how this works or am I doing something wrong?

**Comment by mobsense** [comment]
When a find request is given without the fields needed to complete the full SK, find will do a query with "begins_with" using the leading static portion of the SK that is defined. This is very efficient as DynamoDB can identify the items directly from the keys.

We don't do a filter based on the type as that would entail reading items and then filtering. You can probably do this by providing your type as a parameter.

What we typically do is have a different leading static portion for the SK. i.e. Order vs OrderItem.

i.e. for the order, use "order#" and the order item would be "item#${orderId}#${id}"

**Reply by AlexStansfield** [reply]
Thanks for the response. Now I understand. The reason we setup our SK's like that was so that we could easily get an order and it's items in one find call. Perhaps it's better if we do that with a GSI instead.
